### PR TITLE
optimize_caching.md : Improve description

### PR DESCRIPTION
**Pull request by Siddhartha1234**
**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: data/projects/optimize_caching.md [hunk 1]
```diff
@@ -9,20 +9,28 @@ Own performance optimization ideas are highly encouraged!
...
-* At least two more cases have been 'solved' in a similar fashion.
+* Documentation and code for the 'section division' for 'per-section caching' has been completed and merged.
+* Code for Git like approach to not explore directories that haven't been modified has been tested and merged.
+* Own idea for performance optimization(If any) has been properly documented and work has been started. 
+  
+
+##### CODING PHASE 3
+
+* Report of performance improvements for all the above mentioned criterias. 
+* All the such cases have been 'solved'.
```
Comment (Udayan12167) [inline data/projects/optimize_caching.md]: Can you clarify which cases this is a bit vague?

File: data/projects/optimize_caching.md [hunk 2]
```diff
@@ -9,20 +9,28 @@ Own performance optimization ideas are highly encouraged!
...
 * The respective code has been properly tested, documented and merged.
 
-##### GSOC 2017 FINAL
+##### CODING PHASE 2
 
-* At least two more cases have been 'solved' in a similar fashion.
+* Documentation and code for the 'section division' for 'per-section caching' has been completed and merged.
+* Code for Git like approach to not explore directories that haven't been modified has been tested and merged.
+* Own idea for performance optimization(If any) has been properly documented and work has been started. 
+
```
Comment (sims1253) [inline data/projects/optimize_caching.md]: In preparation and phase 1 no specific methods were mentioned and the student suddenly has to implement exactly these things. These need to be mentioned earlier. Maybe also Update the text above the milestones as well as the bonding and phase 1 milestones as well?

File: data/projects/optimize_caching.md (hunk 1 above)
Comment (sims1253) [inline data/projects/optimize_caching.md]: which cases are these. Again the spontanious arrival of specific goals. Mention them earlier as before :)

Comment (sims1253) [inline data/projects/optimize_caching.md]: Also the last line can be removed and swap solved with implemented. All code that gets put into coala has to be documented and tested completely :D

File: data/projects/optimize_caching.md (hunk 2 above)
Comment (Udayan12167) [inline data/projects/optimize_caching.md]: Extra new line

File: data/projects/optimize_caching.md [hunk 3]
```diff
@@ -9,20 +9,28 @@ Own performance optimization ideas are highly encouraged!
...
 of the above cases.
 * The respective code has been properly tested, documented and merged.
 
-##### GSOC 2017 FINAL
+##### CODING PHASE 2
 
-* At least two more cases have been 'solved' in a similar fashion.
+* Documentation and code for the 'section division' for 'per-section caching' has been completed and merged.
+* Code for Git like approach to not explore directories that haven't been modified has been tested and merged.
+* Own idea for performance optimization(If any) has been properly documented and work has been started.
```
Comment (RohanVB) [inline data/projects/optimize_caching.md]: If you're still using this after the corrections sims suggested, do:
`optimization<space>(if any)`

**Comment by sims1253** [comment]
was cherry-picked

**Pull request closed without merging**